### ISSUE TITLE
Indent relative to block with multiline method chains

### ISSUE THREAD

**Issue by mlwyatt** [rubocop/rubocop#12297]
## Is your feature request related to a problem? Please describe.

While correct when one thinks indent "relative to statement", the following code reads easier if indented "relative to block start"

```ruby
# Relative to statement
data = {
  users:
    User
      .where(id: id_list)
      .group_by(&:last_name)
      .transform_values do |v|
      v.map(&:as_json)
    end
}

# Relative to block
data = {
  users:
    User
      .where(id: id_list)
      .group_by(&:last_name)
      .transform_values do |v|
        v.map(&:as_json)
      end
}
```

## Describe the solution you'd like

Unless there's already a config option I can set that I have yet been able to find, either a new cop or a new option for an existing cop to enforce either style would be great

## Describe alternatives you've considered

I've searched the issue list for `indent` and couldn't find anything. In this particular simple example, I can change the `do` block to use `{}`, but some block will be too long or multiple statements

```ruby
data = {
  users:
    User
      .where(id: id_list)
      .group_by(&:last_name)
      .transform_values { |v| v.map(&:as_json) }
}
```

Any help would be greatly appreciated

**Comment by koic** [comment]
It's likely a false negative for `Layout/IndentationWith` cop.

**Comment by piradata** [comment]
same bug here, it is changing this:

```ruby
def validate_required_configs
  self.class.schema.select { |_name, config| config[:required] }
    .each do |field_name, _config|
      field_name_label = "#{field_name}_label"
      error_message = "#{::I18n.t(field_name_label)} #{::I18n.t('activerecord.errors.messages.blank')}"
      errors.add(:base, error_message) if config[field_name].blank?
  end
end
```

for this:

```ruby
def validate_required_configs
  self.class.schema.select { |_name, config| config[:required] }
    .each do |field_name, _config|
    field_name_label = "#{field_name}_label"
    error_message = "#{::I18n.t(field_name_label)} #{::I18n.t('activerecord.errors.messages.blank')}"
    errors.add(:base, error_message) if config[field_name].blank?
  end
end
```

instead of this:

```ruby
def validate_required_configs
  self.class.schema.select { |_name, config| config[:required] }
    .each do |field_name, _config|
      field_name_label = "#{field_name}_label"
      error_message = "#{::I18n.t(field_name_label)} #{::I18n.t('activerecord.errors.messages.blank')}"
      errors.add(:base, error_message) if config[field_name].blank?
    end
end
```

here are my Layout cops:

```yaml
Layout/HeredocArgumentClosingParenthesis:
  Enabled: false
Layout/LineLength:
  Enabled: true
  Max: 120
Layout/EndOfLine:
  EnforcedStyle: lf
Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
Layout/FirstArgumentIndentation:
  EnforcedStyle: consistent
Layout/MultilineAssignmentLayout:
  EnforcedStyle: same_line
Layout/EndAlignment:
  EnforcedStyleAlignWith: variable
```

**Comment by piradata** [comment]
> It's likely a false negative for `Layout/IndentationWith` cop.

Do you meant `IndentationWidth`? If so then yes, its possible, because it cries about that when I disable the autoformat

![image](https://github.com/rubocop/rubocop/assets/16529503/05b63661-3e50-495f-b122-2acf022ee97f)

And it thinks its all good if I adjust the last `end`...

![image](https://github.com/rubocop/rubocop/assets/16529503/eda0c21c-0aeb-426e-b0df-41c791712ca0)
